internal/worker/removal/worker.go

```diff
@@ -62,8 +85,48 @@ func NewWorker(config Config) (worker.Worker, error) {
 }
 
 func (w *removalWorker) loop() (err error) {
-	<-w.catacomb.Dying()
-	return w.catacomb.Err()
+	watch, err := w.cfg.RemovalService.WatchRemovals()
+	if err != nil {
+		return errors.Capture(err)
+	}
+
+	if err := w.catacomb.Add(watch); err != nil {
+		return errors.Capture(err)
+	}
+
+	ctx := w.catacomb.Context(context.Background())
```

Comment:
The current pattern is to create a scopedContext method off the worker.

https://github.com/juju/juju/blob/357f3a3511c18b5545b5dd107b81eb7e4b143599/internal/worker/logsink/worker.go#L276-L282

The cancel callback is important, as you want to cancel that when you go and query the database (which isn't happening in this PR).

Reply:
As discussed, unless you need to cancel at a specific place other than worker exit, you don't need the cancel callback. The catacomb's tomb created a cancellable context under the hood, which is cancelled when it exits.